### PR TITLE
Update README.extern

### PR DESCRIPTION
**Pull request by mppf**
* restructured the document to better cover the options
  (extern block vs extern decls, command line vs require).
* made various minor improvements to exposition
* cover .c_str() when using extern functions
* cover require as an alternative to listing on command line
* cover renaming functions and types

### CODE REVIEW

File: doc/release/technotes/README.extern.rst [hunk 1]
```diff
@@ -756,6 +764,167 @@ which demonstrates full integration between Chapel and the C function
...
+    .. code-block:: sh
+
+       chpl foo.h foo.c myProgram.chpl
+       chpl foo.h foo.o myProgram.chpl #if foo.c had already been compiled)
+       chpl foo.h -lfoo myProgram.chpl #if foo.c had been archived in libfoo.a)
+
+    Note that you can use -I and -L arguments for the Chapel compiler
+    to specify include or library paths as with a C compiler.
+
+  2) Alternatively, the required C resources can be listed within the
```
Comment (benharsh) [inline doc/release/technotes/README.extern.rst]: I think this is slightly more indented than the previous bullet "1)", is that intentional?

**Comment by benharsh** [comment]
Looks good otherwise, :+1:

File: doc/release/technotes/README.extern.rst (hunk 1 above)
Comment (mppf) [inline doc/release/technotes/README.extern.rst]: Nope, will fix.

File: doc/release/technotes/README.extern.rst [hunk 2]
```diff
@@ -756,6 +764,167 @@ which demonstrates full integration between Chapel and the C function
...
+    .. code-block:: chapel
+
+       require "foo.h"
+       require "foo.c"
+
+
+    This has an effect similar to adding foo.h and foo.c to the
+    command line. You might need to specify -I and -L arguments
+    to indicate to the directories storing any headers or library
+    files.
```
Comment (bradcray) [inline doc/release/technotes/README.extern.rst]: Hi @mppf:  Just happened to glance at this and had a few notes:

* 'require' statements need a final semicolon
* they can also be comma-separated (i.e., one would probably say:

    require "foo.h", "foo.c";

* You can also do require statements of the -l form:

    require "-lfoo";

**Pull request merged**